### ISSUE TITLE
[docs]: Persisting view and viewmodel from extended navigator

### ISSUE THREAD

**Issue by TobyDevelopment** [Stacked-Org/stacked#994]
### Describe the missing piece of documentation

Hi, I have a bottom navigation bar set up with IndexTrackingViewModel. One of the tabs is an extended navigator with some nested navigation. This is all working.

However, when I switch tabs from the nested navigation and then go back to the extended navigator tab/nested navigation, the initial view/route is loaded and the viewmodel onready is called again. How do I persist the view model and view of the extended navigator. How do I load the details view rather than the initial view when going back to that tab.

I have seen stuff using key: PageStorageKey('storage-key') to persist the view state. This works if the page is not in an extended navigator.

Also viewmodel seems to be created every time even when overriding disposeViewModel and fireOnViewModelReadyOnce on tab change, whether the view is in an extendednavigator or not.

TLDR: I want to persist each view and view model or complete navigation stack (if it is nested) in each tab of the bottom nav bar. When tapping the tab it should bring you to the last page it was on within that nav stack. If possible tapping again could pop the entire stack back to start. Can I do this with stacked?

I have searched a lot but cannot find any information on this. I am by no means a flutter expert. Let me know if you need any further info.

Really appreciate the help.

Cheers

**Comment by TobyDevelopment** [comment]
I managed to implement what I wanted by using the AutomaticKeepAliveClientMixin

**Comment by Chappie74** [comment]
@TobyDevelopment  can you give an example please.
